lib/cyrusdb_twom.c

```diff
@@ -0,0 +1,407 @@
+/* cyrusdb_twom.c - wrapper around the twom library
+ *
+ * Copyright (c) 1994-2025 Carnegie Mellon University.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. The name "Carnegie Mellon University" must not be used to
+ *    endorse or promote products derived from this software without
+ *    prior written permission. For permission or any legal
+ *    details, please contact
+ *      Carnegie Mellon University
+ *      Center for Technology Transfer and Enterprise Creation
+ *      4615 Forbes Avenue
+ *      Suite 302
+ *      Pittsburgh, PA  15213
+ *      (412) 268-7393, fax: (412) 268-7395
+ *      innovation@andrew.cmu.edu
+ *
+ * 4. Redistributions of any form whatsoever must retain the following
+ *    acknowledgment:
+ *    "This product includes software developed by Computing Services
+ *     at Carnegie Mellon University (http://www.cmu.edu/computing/)."
+ *
+ * CARNEGIE MELLON UNIVERSITY DISCLAIMS ALL WARRANTIES WITH REGARD TO
+ * THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS, IN NO EVENT SHALL CARNEGIE MELLON UNIVERSITY BE LIABLE
+ * FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+ * AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
+ * OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <config.h>
+
+#include <errno.h>
+#include <libgen.h>
+#include <syslog.h>
+#include <sys/mman.h>
+
+#include "bsearch.h"
+#include "cyrusdb.h"
+#include "libcyr_cfg.h"
+#include "twom.h"
+#include "util.h"
+#include "xmalloc.h"
+
+/* type aliases */
+#define LLU long long unsigned int
+#define LU long unsigned int
```

Comment:
`LU` is nowhere used.

Reply:
I'm pretty sure it is:
```
    printf("HEADER: v=%lu g=%llu fl=%08lX num=(%llu/%llu) sz=(%08llX/%08llX/%08llX) ml=%lu\n",
          (LU)header->version,
          (LLU)header->generation,
          (LU)header->flags,
          (LLU)header->num_records,
          (LLU)header->num_commits,
          (LLU)header->dirty_size,
          (LLU)header->current_size,
          (LLU)header->repack_size,
          (LU)header->maxlevel);
```

Reply:
I meant `LU` is defined in libcyrus_twom.c, but is not used in libcyrus_twom.c.